### PR TITLE
Operator Descriptor Cleanup + Documentation

### PR DESCRIPTION
**Pull request by alecmerdler**
### Description

Code cleanup and documentation in preparation for contributions to spec/status/action descriptors for OLM-managed Operators.

### Changes

- renames `components/cloud-services` to `components/operator-lifecycle-manager` (for clarity)
- tightens up imports/exports for `components/operator-lifecycle-manager/descriptors` 
- adds `operator-lifecycle-manager/descriptors/README.md` with contribution guidelines

### Next Steps

- Implement adding `actionDescriptors` to cog menus
- Read annotations on CSV for descriptor component positioning
- Write e2e test which creates CSV will all descriptor components

Addresses https://jira.coreos.com/browse/ALM-687

### CODE REVIEW

File: frontend/public/components/operator-lifecycle-manager/descriptors/README.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# Operator Descriptors
```
Comment (alecmerdler) [inline frontend/public/components/operator-lifecycle-manager/descriptors/README.md]: First draft of instructions for contributing descriptor components.

File: frontend/public/components/operator-lifecycle-manager/descriptors/spec/configure-size.tsx [hunk 2]
```diff
@@ -13,19 +13,12 @@ export const configureSizeModal = ({kindObj, resource, specDescriptor, specValue
     message: specDescriptor.description,
     path: `/spec/${specDescriptor.path}`,
     buttonText: `Update ${specDescriptor.displayName}`,
-    invalidateState: (isInvalid) => {
-      // NOTE: Necessary until https://github.com/kubernetes/kubernetes/pull/53345 fixes WebSocket loading of the custom resources.
```
Comment (alecmerdler) [inline frontend/public/components/operator-lifecycle-manager/descriptors/spec/configure-size.tsx]: This PR was merged!

**Comment by alecmerdler** [comment]
jenkins rebuild

File: frontend/public/components/operator-lifecycle-manager/descriptors/README.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    - kind: Service
+      version: v1
+    - kind: Pod
+      version: v1
+  specDescriptors:
+    - description: The desired number of member Pods for the etcd cluster.
+      displayName: Size
+      path: size
+      x-descriptors:
+        - 'urn:alm:descriptor:com.tectonic.ui:podCount'
```
Comment (josephschorr) [inline frontend/public/components/operator-lifecycle-manager/descriptors/README.md]: Do we want to take this opportunity to deprecate `alm` and replace it with `olm`? We'd still need to support `alm` for back-compat, but we could indicate to users not to use it

File: frontend/public/components/operator-lifecycle-manager/descriptors/status/index.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import * as _ from 'lodash-es';
+import { Map as ImmutableMap } from 'immutable';
+
+import { ResourceLink } from '../../../utils';
+import { StatusCapability, CapabilityProps, DescriptorProps } from '../types';
+import { Phase } from './phase';
+import { PodStatusChart } from './pods';
+
+// TODO(alecmerdler): Move these to separate module?
+const Default: React.SFC<StatusCapabilityProps> = ({value}) => _.isEmpty(value) && !_.isNumber(value)
```
Comment (josephschorr) [inline frontend/public/components/operator-lifecycle-manager/descriptors/status/index.tsx]: We should add a block comment before each of these describing how to instantiate (its full name) and what it does

**Comment by alecmerdler** [comment]
jenkins rebuild

**Comment by alecmerdler** [comment]
jenkins rebuild

**Comment by alecmerdler** [comment]
jenkins rebuild

**Pull request merged**